### ISSUE TITLE
[IMAP] - Auth token missing for this account

### ISSUE THREAD

**Issue by diegomachadosoares** [nylas-mail-lives/nylas-mail#107]
**Are there any related issues?**

Not that I could find on the search... maybe #51.

**What operating system are you using?**

ArchLinux (I'll try to reproduce on OSX).

**What version?**

built from master branch

**Do you have any third-party plugins installed? If so, which ones?**

Nop. Just the inbox theme.

**Is the issue related to a specific email provider (Gmail, Exchange, etc.)?**

Yes. Just on my private IMAP server account. On GApps it seems to be working fine.

**Issue**

My IMAP account was woking fine until I added my GApps account (through IMAP, not GMail Oauth). Although it continues to update my inbox, it doesn't update the message status (read, deleted, etc) and doesn't save sent messages as well giving me this error message:

 "ERROR We were unable to apply the changes to your thread, please try again later.

Error message: Auth token missing for account"

I could trace the message back [here]( https://github.com/nylas-mail-lives/nylas-mail/blob/master/packages/client-app/src/flux/nylas-api-request.es6#L143-L147), but haven't been able to go much further...

**Comment by dweremeichik** [comment]
I wonder if it expects there to be a token because there is a Gmail account. Perhaps related to allowing the google account without Oauth: https://github.com/nylas-mail-lives/nylas-mail/commit/18545f3add6cefe2af36ae80f5dd427bb1752ad2

**Comment by diegomachadosoares** [comment]
Hi, @dweremeichik. 

With GApps (through simple IMAP exactly as the other) it is working fine. The problem is with other account (work account).

Also there is a red square saying that "Cannot Authenticate Nylas Mail Cloud Services with xxx@xxx.com" which is strange since it IS updating the inbox of the problematic account with new messages... And all contact with the non-local "Nylas Cloud" (i.e. n1.nylas.com) is blocked through firewall, so if it was the problem, the GApps account would not be working as well and the message would say the same thing for the GApps as well.

![nylas2](https://user-images.githubusercontent.com/5024553/31311930-7f262b62-ab8d-11e7-9ad6-754eb107e77a.png)
Regards,

**Comment by simonft** [comment]
@diegomachadosoares did you get a password management error when you first logged in? I think it might be related to #51

**Comment by diegomachadosoares** [comment]
No, @simonft! I thought that too... the curious thing is: I added another account from the same domain/provider and it works fine... But there are other strange things happening as well... On the GApps account new messages appear for a moment and then disappear! I checked on the webmail and they are there, but doesn't appear on Nylas anymore...

**Comment by dweremeichik** [comment]
I think I figured out a little more about this. I believe that this is the Nylas API token. Which should normally exist in a table in shared.sqlite (don't remember the table name, there are only like 4 tables though). Perhaps you can check to see if the token is missing? There should be 1 token per account.

**Comment by dweremeichik** [comment]
Don't know why I didn't realize this earlier, the file you traced back to in your intial post is "nylas-api-request".

**Comment by simonft** [comment]
I figured that an API token was gone, but I wasn't sure which. Which API token are you thinking? There's one for the local API, and one for the Nylas Cloud API. I removed things that should need the Cloud API token, but maybe I messed that up. It's also possible the local API token isn't getting stored correctly for some reason.

**Comment by diegomachadosoares** [comment]
@dweremeichik and @simonft, there are 4 tokens in the table... The first (ID=1) I believe that is the Nylas Cloud API token (was generated in April so, can't be one of mine). And there is the other 3 (of my tree accounts (private provider, GApps, private provider, in this order).
![nylas-db](https://user-images.githubusercontent.com/5024553/31389957-f261103e-ada8-11e7-954e-9c83772c309e.png)

**Comment by diegomachadosoares** [comment]
I removed the ".nylas-mail", reinstalled the app, and added the private provider account (no other) and the problem persists... but now I only have one token in the table (generated today). So there is a token problem.... but the strange thing is that the account IS synchronizing...

**Comment by diegomachadosoares** [comment]
Seems to have fixed! But still need some testing...

Here is the procedure I used.

1 - Removed the ".nylas-mail"
2 - Reinstalled the app (with the latest PRs, from master)
3 - Added the private provider account (no other) 
Result: Problem persists... 

Second try

1 - Removed the ".nylas-mail"
3 - Added GApps accounts FIRST then the others
Result: Fixed! Everything seems to be working. I'll make some more testing and post the results...

**Comment by l2dy** [comment]
Reproduced on macOS, only one IMAP account, fresh install of `nylas-mail-2.2.2.dmg`.

![screen shot](https://user-images.githubusercontent.com/14329097/32732339-28184cba-c884-11e7-97e6-a9b5f150144e.png)

**Comment by diegomachadosoares** [comment]
Hi, @l2dy. Is this a GMail or GApps account or some other provider?

**Comment by l2dy** [comment]
@diegomachadosoares Other provider.
